### ISSUE TITLE
Allow support for scrubbing sensitive data from tapes

### ISSUE THREAD

**Issue by tmtrademarked** [airbnb/okreplay#60]
We're using OkReplay to write UI tests, and it's been working well for us so far! But one problem we've had with the library is that the generated tapes sometimes contain sensitive information like passwords/tokens/etc which we would prefer not to check in.

One solution that feels reasonable would be to allow us to create a TapeRoot which implements a Reader/Writer that knows how to replace sensitive data with safe tokens and vice versa. Unfortunately, AndroidTapeRoot is final, and the PermissionRule has a hard requirement of an AndroidTapeRoot instance.

Is there a better solution available for scrubbing out sensitive data before storing the tape? Alternately, could we just make AndroidTapeRoot non-final? Thoughts?

**Comment by felipecsl** [comment]
Great point! I think it would make sense to make `AndroidTapeRoot` non-final if that would resolve your immediate issue. I'll send a quick PR

**Comment by slott** [comment]
@tmtrademarked did you manage to get something working in terms of scrubbing the tapes ?
I'm in the exact same boat that you were in - and it would be nice if I didn't have to start from scratch.

**Comment by tmtrademarked** [comment]
Yes, I did - although it's not necessarily the most generic solution. I'm
out of the office until next week, but I'll try to at least post something
here then until I can get a pull request put together.
